From 8be0f5b012dc8d0637a0f1664b556538640439c6 Mon Sep 17 00:00:00 2001
From: 1vivy <1vivy@tutanota.com>
Date: Tue, 21 Jul 2026 23:46:34 -0600
Subject: [PATCH 21/22] sepolicy: address Oplus touch sys_nice + default_prop
 denials

Touch AIDL HAL bumps its own RT priority (self:capability sys_nice) and
harmlessly probes default_prop; ufcs_dev + persist-dir perms are already
covered by our tree. Incremental gap over c74df81f.

(cherry picked from commit ee1ff33ac79a33fc5630e30e0958c3c3b9c0255c)
---
 sepolicy/qti/vendor/hal_oplus_touch_aidl.te | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/sepolicy/qti/vendor/hal_oplus_touch_aidl.te b/sepolicy/qti/vendor/hal_oplus_touch_aidl.te
index 156d6faf..63fe7c2e 100644
--- a/sepolicy/qti/vendor/hal_oplus_touch_aidl.te
+++ b/sepolicy/qti/vendor/hal_oplus_touch_aidl.te
@@ -21,6 +21,7 @@ allow hal_oplus_touch_aidl input_device:dir search;
 allow hal_oplus_touch_aidl input_device:chr_file rw_file_perms;
 
 allow hal_oplus_touch_aidl oplus_touchdaemon_device:chr_file rw_file_perms;
+allow hal_oplus_touch_aidl self:capability sys_nice;
 
 allow hal_oplus_touch_aidl proc_bus_input:file r_file_perms;
 allow hal_oplus_touch_aidl proc_horae:file r_file_perms;
@@ -32,3 +33,5 @@ rw_dir_file(hal_oplus_touch_aidl, vendor_sysfs_usb_supply)
 
 set_prop(hal_oplus_touch_aidl, system_oplus_touch_prop)
 set_prop(hal_oplus_touch_aidl, vendor_oplus_touch_prop)
+
+dontaudit hal_oplus_touch_aidl default_prop:file read;
